test/KregielniaTests.cpp

```diff
@@ -43,3 +43,19 @@ TEST_F(KregielniaTests, assertExampleConversion)
     EXPECT_EQ(sumVector(ConvertStringToVector("X|4-|3")), 17);
     EXPECT_EQ(sumVector(ConvertStringToVector("34|X|0-")), 17);
 }
+
+TEST_F(KregielniaTests, assertCorrectNameExtraction)
+{
+    EXPECT_EQ(GetPlayerName("Name1:X|4-|3"), "Name1");
+    EXPECT_EQ(GetPlayerName(":X|22|33"), "");
+    EXPECT_EQ(GetPlayerName(":"), "");
+    EXPECT_EQ(GetPlayerName(""), "");
```

Comment:
What if the name contains a `:` sign? :)
Please add such a testcase and assert that it also work as expected.

Reply:
Done

Reply:
Function looks for last of ":".